tests/models/test_data/test_bad_structures.json

```diff
@@ -1503,7 +1503,13 @@
       "O",
       "O"
     ],
-    "structure_features": ["assemblies"],
```

Comment:
c.f. comment above, we can no longer detect this as a problem as the "assemblies" field can now be missing or null

Reply:
That's fine, I guess?
Although, we should probably check the truth of this value still, i.e., _if_ `"assemblies"` is included in `"structure_features"`, then the key `assemblies` MUST be present ...
And vice versa: If `assemblies` is present, `structure_features` MUST include `"assemblies"` (as stated in the spec).

Reply:
That _should_ be handled as the validators now as they are

Reply:
No issue then. That means that we _can_ detect if this is a problem, I presume, and your original comment is wrong?

Reply:
We can still detect inconsistency between the two fields if provided, but it is now valid for `structure_features` to contain assemblies, whilst assemblies is not present/`null` (which would be the same as if there are no assemblies in the structure).

I'm a little bit worried about the special treatment `structure_features` now gets as the only "mandatory" field, there's now an artificial distinction between allowing `null` values because they are "SHOULD" versus allowing null values because you don't have to provide them if `response_fields` is set in a particular way...

Reply:
> We can still detect inconsistency between the two fields if provided, but it is now valid for `structure_features` to contain `"assemblies"`, whilst the `assemblies` field is not being present/`null` (which would be the same as if there are no assemblies in the structure).

Why? I don't understand. Why couldn't we just check that if `"assemblies"` is part of the value for `structure_features` then that subsequently means `assemblies` MUST be present...?

> I'm a little bit worried about the special treatment `structure_features` now gets as the only "mandatory" field, there's now an artificial distinction between allowing `null` values because they are "SHOULD" versus allowing null values because you don't have to provide them if `response_fields` is set in a particular way...

Well `id` and `type` are still mandatory - but they're not technically properties I guess ...
And I see your point. But I don't think the distinction will come up in practical use cases?

Reply:
> Why? I don't understand. Why couldn't we just check that if `"assemblies"` is part of the value for `structure_features` then that subsequently means `assemblies` MUST be present...?

The spec says that support for the `assemblies` field is optional, and doesn't mandate that it MUST be present if `structure_features` contains `"assemblies"` (as we were previously checking), but there is no explicit statement the other way around. I think I agree with you now that we should keep the old check though, even if its not explicitly stated... 

> Well `id` and `type` are still mandatory - but they're not technically properties I guess ...
> And I see your point. But I don't think the distinction will come up in practical use cases?

Hopefully not!

Reply:
My main annoyance here is that we can't include `assemblies` in the combinatorial `null` tests, as there is one edge-cases where an optional field _must_ be included based on the value of another field, which is probably why I removed the check...